### PR TITLE
GEODE-2997: Change new protocol GetAllResponse.

### PR DESCRIPTION
**Pull request by galen-pivotal**
@kohlmu-pivotal @hiteshk25 @WireBaron @pivotal-amurmann @bschuchardt 

I think it's better to return the errors and the successful keys so that
one failed lookup doesn't mess up the whole request.

Imagining this as a conversational PR to see what the response is.

Signed-off-by: Galen O'Sullivan <gosullivan@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [X] Is your initial contribution a single, squashed commit?

- [NO.] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by galen-pivotal** [comment]
Some other thoughts from today:
* make GetAllResponse return a collection of Entries and a collection of errors.
* make Get and GetAll return the same type of response, as mentioned above. Get will always return a single key in either the success or error collections.
   - maybe looking in a collection is extra overhead for a simple Get request?
   - bigger errors (like malformed messages) will still return errors at the Response level rather than errors nested in the response message.

Thoughts?

File: geode-protobuf/src/main/proto/basicTypes.proto [hunk 1]
```diff
@@ -62,4 +62,14 @@ message Region {
 
 message Server {
     string url = 1;
-}
\ No newline at end of file
+}
+
+message Error {
```
Comment (WireBaron) [inline geode-protobuf/src/main/proto/basicTypes.proto]: These are the same fields as ErrorResponse and KeyedErrorResponse in the outstanding PutAll pull request, can you comment in that PR to change the names if you prefer these?

File: geode-protobuf/src/main/proto/clientProtocol.proto [hunk 2]
```diff
@@ -66,7 +66,7 @@ message Response {
         RemoveAllResponse removeAllResponse = 7;
         ListKeysResponse listKeysResponse = 8;
 
-        ErrorResponse errorResponse = 13;
+        Error error = 13;
```
Comment (kohlmu-pivotal) [inline geode-protobuf/src/main/proto/clientProtocol.proto]: Why has this been renamed to Error from ErrorResponse? Imo, we are following the pattern (good or bad) that a ClientProtocol.Response contains an Operation specific response. To now have a non-"Response" message sort of breaks the mold.

File: geode-protobuf/src/main/proto/basicTypes.proto [hunk 3]
```diff
@@ -62,4 +62,14 @@ message Region {
...
+    int32 errorCode = 1;
+    string message = 2;
+}
+
+message ErrorEntry {
```
Comment (kohlmu-pivotal) [inline geode-protobuf/src/main/proto/basicTypes.proto]: Tbh, I'm not ecstatic about the name. I'd prefer `FailedEntry`

File: geode-protobuf/src/main/proto/clientProtocol.proto (hunk 2 above)
Comment (galen-pivotal) [inline geode-protobuf/src/main/proto/clientProtocol.proto]: The reasoning is, we're not just using it as an ErrorResponse at the same level as a PutResponse or GetResponse -- it gets embedded in other responses now.

File: geode-protobuf/src/main/proto/basicTypes.proto (hunk 3 above)
Comment (galen-pivotal) [inline geode-protobuf/src/main/proto/basicTypes.proto]: That's fine too. I don't have a real strong preference.

File: geode-protobuf/src/main/proto/clientProtocol.proto (hunk 2 above)
Comment (kohlmu-pivotal) [inline geode-protobuf/src/main/proto/clientProtocol.proto]: To keep the "Response -> OperationalResponse" hierarchy, I'd prefer to keep "ErrorResponse" that contains an "Error" object.
We _could_ inline it, but it detracts from the modus-operandi

**Comment by galen-pivotal** [comment]
This isn't code complete; I'm planning to work on this when I have the time (unless anyone else wants to).

**Comment by kohlmu-pivotal** [comment]
@galen-pivotal is this PR active anymore?

**Comment by galen-pivotal** [comment]
closing for #739 .

**Pull request closed without merging**